### PR TITLE
multi: Update dcrd and dcrwallet dependencies

### LINKED ISSUE #135: Update `docker/README.md` - currently it leads to an error
### Background

I tried to go through `docker/README.md` today and came across this issue

### Your environment

* Version of `dcrlnd`
master
* Which operating system (`uname -a` on *Nix).
linux
* Version of `dcrd` or other backend.
Dockerfile pinned 1.5.1

### Steps to reproduce

`docker-compose run -d --name alice dcrlnd` :

```
2021-03-07 18:19:19.683 [DBG] DCRW: Starting rpc syncer
2021-03-07 18:19:19.732 [INF] DCRW: Headers synced through block 6bef82c645999585f7255cb02672921ac2f5492820090cd635fe3a59d16b4f87 height 0
2021-03-07 18:19:19.732 [INF] DCRW: Transactions synced through block 6bef82c645999585f7255cb02672921ac2f5492820090cd635fe3a59d16b4f87 height 0
2021-03-07 18:19:19.733 [ERR] DCRW: RPCSyncer error: rpcsyncer.Run:
        dcrd.GetMempoolTSpends: Invalid transaction type: tspend -- supported types: [regular tickets votes revocations all]            
```

### Expected behaviour

Maybe the dcrlnd version should be pinned to a tag without TSPEND or the dcrd/dcrwallet Dockerfiles should be updated to 1..6.1. I'm unsure if this should wait until the TSPEND fork happens

### PR DESCRIPTION
**Pull request by matheusd**
This updates the dcrd and dcrwallet dependencies to their current master versions.

It also bumps the current minor version to v0.4.0.

It also updates and fixes the docker requirements. close #135

### CODE REVIEW

**Review by jholdstock — APPROVED** [review]
LGTM. Nice to see the benefits of the `dcrtest` package played out

**Review by dajohi — APPROVED** [review]

**Review by alexlyp — APPROVED** [review]

File: docker/dcrd/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,12 @@
-FROM golang:1.15-alpine as builder
+FROM golang:1.19-alpine as builder
```
Comment (jholdstock) [inline docker/dcrd/Dockerfile:1]: 1.20?

File: lnwallet/remotedcrwallet/extended_wallet.go [hunk 2]
```diff
@@ -43,3 +46,28 @@ func (b *DcrWallet) RescanWallet(startHeight int32, progress func(height int32)
...
+	}
+	res, err := b.wallet.GetTransaction(b.ctx, req)
+	if status.Code(err) == codes.NotFound {
+		return nil, lnwallet.ErrWalletTxNotExist
+	}
+	if err != nil {
+		return nil, err
+	}
+	var bh *chainhash.Hash
+	if res.BlockHash != nil {
```
Comment (jholdstock) [inline lnwallet/remotedcrwallet/extended_wallet.go:62]: Why would BlockHash be nil? Is it if the tx is not yet mined?

Comment (matheusd) [reply lnwallet/remotedcrwallet/extended_wallet.go:62]: Correct.

**Pull request merged**